src/table/sorting.js

```diff
@@ -1,11 +1,17 @@
 export default function sortingFactory(model) {
-  return (layout, isDim, sortIdx) => {
+  return (layout, isDim, label) => {
+    const { qDimensionInfo, qMeasureInfo } = layout.qHyperCube;
+    const index = isDim
+      ? qDimensionInfo.findIndex((dimensionInfo) => dimensionInfo.qFallbackTitle === label)
+      : qMeasureInfo.findIndex((measureInfo) => measureInfo.qFallbackTitle === label);
+    const idx = isDim ? index : index + qDimensionInfo.length;
```

Comment:
This is exactly what qColumnOrder can be used for. `qColumnOrder[sortIdx]` should give you the same thing.

Reply:
qColumnOrder is the order of columns
<img width="231" alt="Screenshot 2021-03-30 at 09 31 22" src="https://user-images.githubusercontent.com/4471489/112950567-b4a86e80-913a-11eb-9428-6c5a500d32f2.png">

idx is the order of sorting
<img width="258" alt="Screenshot 2021-03-30 at 09 32 01" src="https://user-images.githubusercontent.com/4471489/112950635-cb4ec580-913a-11eb-9e14-887f5471be69.png">

Reply:
but when the order of Columns is not set, like the case in nebula, qColumnOrder is []. qColumnOrder[sortIdx] will be undefined.

Reply:
ah true, but then `getColumnOrder` returns a proper array

Reply:
return that with `manageData` then send it into `sortingFactory`. Also I'd run `sortingFactory` inside the render `useEffect`.

Reply:
Right. Good idea. I Will work on it.